src/middleware/devtools.ts

```diff
@@ -294,66 +294,70 @@ export function devtools<
       }
     }
 
-    extension.subscribe((message: any) => {
-      switch (message.type) {
-        case 'ACTION':
-          if (typeof message.payload !== 'string') {
-            console.error(
-              '[zustand devtools middleware] Unsupported action format'
-            )
-            return
-          }
-          return parseJsonThen<{ type: unknown; state?: PartialState<S> }>(
-            message.payload,
-            (action) => {
-              if (action.type === '__setState') {
-                setStateFromDevtools(action.state as PartialState<S>)
-                return
+    ;(extension as any) // FIXME no-any
```

Comment:
It would be nice if we could use this type definition: https://github.com/reduxjs/redux-devtools/blob/b82de745928211cd9b7daa7a61b197ad9e11ec36/extension/src/app/api/index.ts#L522-L536
```ts
export interface ConnectResponse {
  init: <S, A extends Action<unknown>>(
    state: S,
    liftedData?: LiftedState<S, A, unknown>
  ) => void;
  subscribe: <S, A extends Action<unknown>>(
    listener: (message: ListenerMessage<S, A>) => void
  ) => (() => void) | undefined;
  unsubscribe: () => void;
  send: <S, A extends Action<unknown>>(
    action: A,
    state: LiftedState<S, A, unknown>
  ) => void;
  error: (payload: string) => void;
}
```

Reply:
Yeah. It's a little weird that `ConnectResponse` in `@redux-devtools/extension` only has `init` and `send`, one could consider opening an issue to write the actual type definition too.

Reply:
filed: https://github.com/reduxjs/redux-devtools/issues/1097